options.go

```diff
@@ -51,6 +51,14 @@ func WithLogger(msg *log.Logger) Option {
 	}
 }
 
+// WithAutomaticReconnect allows to configure a socket to automatically
+// reconnect on connection loss.
+func WithAutomaticReconnect(automaticReconnect bool) Option {
```

Comment:
could you please add a test (or example) that tries this new API out?

Reply:
I added a test, but I am not completely happy with it yet, it will look cleaner once #128 is merged and the the resusbscribe is not necessary anymore